lib/px/utils/pxlogin_credentials.lua

```diff
@@ -86,6 +85,93 @@ function M.load(px_config)
         return _M.creds_extract_from_table(ci, body_json)
     end
 
+    local function split(s, sep)
+        local fields = {}
+        local sep = sep or " "
+        local pattern = string.format("([^%s]+)", sep)
+        string.gsub(s, pattern, function(c) fields[#fields + 1] = c end)
+        return fields
+    end
+
+    -- parse lines similar to:  form-data; name1=val1; name2=val2
+    local function decode_content_disposition(value)
+        local result
+        local disposition_type, params = string.match(value, "([%w%-%._]+);(.+)")
+        if disposition_type then
+            result = {}
+            if params then
+                for index, param in pairs(split(params, ";")) do
+                    key, value = param:match('([%w%.%-_]+)="(.+)"$')
+                    if key then
+                       result[key] = value
+                    end
+                end
+            end
+        end
+
+        return result
+    end
+
+    function _M.creds_extract_from_formdata(ci)
+        local chunk_size = 4096
+        local form, err = upload:new(chunk_size)
+        if not form then
+            return nil
+        end
+
+        local user = nil
+        local pass = nil
+
+        form:set_timeout(1000)
```

Comment:
What is this timeout for?

Reply:
This is the maximal time POST body reading could take.